Flapjack/CoreData/CoreDataSource.swift

```diff
@@ -137,13 +136,20 @@ public class CoreDataSource<T: NSManagedObject & DataObject>: NSObject, NSFetche
 
         do {
             Logger.debug("Fetching cache key \"\(cacheKey)\"")
+            controller.delegate = self
             try controller.performFetch()
             hasExecuted = true
         } catch let error {
             Logger.debug("Error fetching CoreDataSource<\(T.self)>: \(error)")
         }
     }
 
+    public func endListening() {
+        guard hasExecuted else { return }
+        controller.delegate = nil
```

Comment:
Even if you `nil` out the controller's `delegate`, is it by chance around trying to use up memory? And do you think we need to call `endListening` inside of a `deinit`, or is that overkill?

Reply:
I think it's overkill, because the fetched results controller should get released, and so will its weak (one would assume) hold on the delegate.  The `nil` ing out of the delegate isn't meant to free up memory as much as prevent the callback methods from being called - I saw some crashes when switching users that made me think that somehow something was still getting held on to, and I wanted a way to definitely stop being notified.